docs/contracts/v4/guides/15-single-sided-liquidity.mdx

```diff
@@ -0,0 +1,423 @@
+
+Positions on a v4 Pool are always created with a Tick range in which their liquidity is accessible to swaps on the Pool.
+
+If we consider the structure of mint function in the Position Manager contract, that helps mint a new position.
+
+```solidity
+function _mint(
+    PoolKey calldata poolKey,
+    int24 tickLower,
+    int24 tickUpper,
+    uint256 liquidity,
+    uint128 amount0Max,
+    uint128 amount1Max,
+    address owner,
+    bytes calldata hookData
+)
+```
+
+We see that a position only stores a single `liquidity` value, and a `tickLower` and `tickUpper` value that define the range in which the liquidity of the Position can be utilised for Swaps.
+The actual **amount** of `token0` and `token1` that a Pool owes the Position owner is calculated from the parts of the liquidity position that are to the left and right of the current Tick.
+Liquidity left of the current Tick is denominated in `token0` and liquidity right of the current Tick is denominated in `token1`.
+
+If a new Position is created and the Tick Range of the position does not include the current Tick of the Pool, only one of the two Tokens in the Pool can be provided.
+
+We will call this a **Single Side Liquidity Position**.
+
+<img src={require('../images/range-order.png').default} alt="RangeOrder" box-shadow="none"/>
+
+When the current Tick of the Pool moves across the Position, the ratio of `token0` and `token1` will change, and ultimately inverse if the current Tick moves out of the position on the other side.
+
+We will utilise this behaviour to provide liquidity with `token1` and withdraw the position when it has been converted to `token0`.
+
+## Calculating the Tick Range
+
+Our goal for this guide is to create a [Buy Limit Order](https://docs.uniswap.org/concepts/protocol/range-orders#buy-limit-orders) that trades `token1` for `token0` when the Price of `token1` decreases by 5%.
+
+To create our Position, we need to first decide the Tick Range that we want to provide liquidity in.
+
+### Upper Tick
+
+We [create a Pool](https://docs.uniswap.org/contracts/v4/quickstart/create-pool) that represents the v4 Pool we are interacting with and get the `token1Price`.
+We won't need full tick data in this example.
+
+```solidity
+import {TickMath} from "v4-core/src/libraries/TickMath.sol";
+import {Constants} from "./base/Constants.sol";
+import {Config} from "./base/Config.sol";
+
+uint24 public constant LP_FEE = 3000;
+int24 public constant TICK_SPACING = 60;
+uint256 public constant token1Amount = 1e6; //USDC
+
+PoolKey memory pool = PoolKey({
+    currency0: currency0, // imported from Config.sol
+    currency1: currency1, // imported from Config.sol
+    fee: LP_FEE,
+    tickSpacing: TICK_SPACING,
+    hooks: hookContract
+});
+
+// Get current pool sqrtPrice and tick
+(uint160 sqrtPriceX96,,,) = POOLMANAGER.getSlot0(pool.toId()); //POOLMANAGER is imported from Constants.sol
+int24 currentTick = TickMath.getTickAtSqrtPrice(sqrtPriceX96);
```

Comment:
this code snippet should be trimmed to the `.getSlot0()` and `.getTickAtSqrtPrice()` function calls; all else feels like bloat, especially the imports